### ISSUE TITLE
@ServerVariable without allowed values not well rendered

### ISSUE THREAD

**Issue by blackat** [swagger-api/swagger-core#3071]
Hello,
I have this description
```java
@OpenAPIDefinition(
        info = @Info(
                title = "My Rest API",
                version = "1.0.0",
                description = "My RESTful API implementation"
        ),
        servers = {
                @Server(
                        description = "My REST API",
                        url = "/{context-path}/{rest-api}",
                        variables = {
                                @ServerVariable(name = "context-path", defaultValue = "my-war"),
                                @ServerVariable(name = "rest-api", defaultValue = "api", allowableValues = {"api", "rest", "batchapi"})
                        }
                )
        }
)
```
I use the `swagger-maven-plugin` version 2.0.6 to generate the JSON in open api 3.0 format.
I obtain this JSON
```javascript
"servers" : [ {
    "url" : "/{context-path}/{rest-api}",
    "description" : "My REST API",
    "variables" : {
      "context-path" : {
        "default" : "my-war",
        "enum" : [ "" ]
      },
      "rest-api" : {
        "default" : "api",
        "enum" : [ "api", "rest", "batchapi" ]
      }
    }
  } ],
```
The problem is the empty `"enum" : [ "" ]` that in the swagger-ui is rendered as an empty dropdown. Without the empty enum there is an input field with the default value so it is possible to set a different context path.

Is there please a workaround?

**Comment by blackat** [comment]
From the source code `io.swagger.v3.jaxrs2.Reader` line 297 
```java
// OpenApiDefinition servers
AnnotationsUtils.getServers(openAPIDefinition.servers()).ifPresent(servers ->openAPI.setServers(servers));
```

Then
```java
public static Optional<List<Server>> getServers(io.swagger.v3.oas.annotations.servers.Server[] servers) {
        if (servers == null) {
            return Optional.empty();
        }
...
}
```
basically it is never null and hence, I think, there is the empty enum `"enum" : [ "" ]`

**Comment by frantuma** [comment]
Thanks for reporting this! fixed in #3071.
